### ISSUE TITLE
Proposal: check for unused //nolint comments

### ISSUE THREAD

**Issue by vearutop** [golangci/golangci-lint#832]
Sometimes the code needs a`//nolint` suppressor to disable check(s), but as the code evolves the `//nolint`may become redundant.

For example such comment may become obsolete when the symbol is actually used:
```go
// MyConst is something special.
// nolint:unused // To be used in subsequent PR.
const MyConst = "my-const"
```

I would like to check that `//nolint` comment was relevant to at least one linting issue and emit lint error otherwise.

**Comment by tpounds** [comment]
@vearutop There was pr #740 that was recently closed to implement similar functionality.

**Comment by tpounds** [comment]
cc @ashanbrown @jirfag

**Comment by jirfag** [comment]
I like this idea and it's not difficult to implement in `nolint` filter

**Comment by ldez** [comment]
`nolintlint` handle this case now:

```
  nolintlint:
    # Enable to ensure that nolint directives are all used. Default is true.
    allow-unused: false
```
